client/init.go

```diff
@@ -0,0 +1,492 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/greenbone/eulabeia/connection"
+	"github.com/greenbone/eulabeia/messages"
+	"github.com/greenbone/eulabeia/messages/cmds"
+	"github.com/greenbone/eulabeia/messages/info"
+	"github.com/greenbone/eulabeia/models"
+	"github.com/rs/zerolog/log"
+)
+
+// Verify and Parse uses init event and received bytes to return bool for finished, messages.Event
+// the parsed bytes or an error
+type VerifyAndParse func(messages.Event, []byte, messages.Message) (bool, messages.Event, error)
+
+// OnPrevious uses the output of a program to create a new init for the current
+type OnPrevious func(messages.Event) messages.Event
+
+// State represents the state of a received message
+type State int8
+
+const (
+	None    = -1 // Not relevant for the program
+	Failure = 0  // Identified as a failure message
+	Success = 1  // Identified as a success message
+)
+
+// Received is used to identify downstream handler about success or failure
+type Received struct {
+	State State
+	Event messages.Event
+}
+
+// Program is a predefined message handler
+type Program struct {
+	sync.RWMutex
+	context           string
+	init              messages.Event
+	verifySuccess     VerifyAndParse
+	verifyFailure     VerifyAndParse
+	onPreviousSuccess OnPrevious
+	onPreviousFailure OnPrevious
+	success           messages.Event
+	failure           messages.Event
+	next              *Program
+	previous          *Program
+	send              chan<- *connection.SendResponse
+	receive           <-chan *connection.TopicData
+	received          chan<- *Received // used to inform downstream about failure or success; mostly useful on multiple success or fialure states (e.g. a scan)
+	timeout           time.Duration    // Timeout of trying to receive a response; a timeout is mandataroy ohterwise it could block
```

Comment:
```suggestion
	received          chan<- *Received // Used to inform downstream about failure or success; mostly useful on multiple success or failure states (e.g. a scan)
	timeout           time.Duration    // Timeout of trying to receive a response; a timeout is mandatory otherwise it could block
```